lib/features/thread/data/repository/thread_repository_impl.dart

```diff
@@ -63,10 +64,8 @@ class ThreadRepositoryImpl extends ThreadRepository {
         accountId,
         limit: limit,
         sort: sort,
-        filter: emailFilter?.filter,
+        filter: EmailFilterCondition(inMailbox: emailFilter?.mailboxId),
```

Comment:
this fixed will wrong for the usecase:
```
- I am logined in tmail
- I open the inbox
- I choose Unread filter
- I go to Mailbox A - which have 23 unread message in page number 2 (index of first unread message is 21)
```

Reply:
demo:

https://user-images.githubusercontent.com/6462404/188813726-4f3f043f-96ba-4b0e-8eab-f691dcbf3877.mov